pyband/wallet/signer.py

```diff
@@ -0,0 +1,84 @@
+from abc import abstractmethod
+
+from ecdsa.der import remove_sequence, UnexpectedDER, remove_integer
+
+from .address import Address
+from .private_key import PrivateKey
+from .public_key import PublicKey
+from ..cosmos_app import CosmosApp
+from ..utils import bip44_to_list
+
+
+class Signer:
+    def __init__(self):
+        self.public_key = self.get_public_key()
+        self.address = self.get_address()
+
+    @abstractmethod
+    def get_public_key(self) -> PublicKey:
+        """Gets public key associated with the signer.
+
+        Returns:
+            A PublicKey instance.
+        """
+        pass
+
+    @abstractmethod
+    def get_address(self) -> Address:
+        """Gets address associated with the signer.
+
+        Returns:
+            An Address instance.
+        """
+        pass
+
+    @abstractmethod
+    def sign(self, msg: bytes) -> bytes:
+        """Signs a message with the signer.
+
+        Args:
+            msg: The message to sign in bytes.
+
+        Returns:
+            Signature in bytes.
+        """
+        pass
+
+
+class PrivateKeySigner(Signer):
+    def __init__(self, private_key: PrivateKey):
+        self.private_key = private_key
+        super().__init__()
+
+    def get_public_key(self) -> PublicKey:
+        return self.private_key.to_public_key()
+
+    def get_address(self) -> Address:
+        return self.public_key.to_address()
+
+    def sign(self, msg: bytes) -> bytes:
+        return self.private_key.sign(bytes(msg))
+
+
+class LedgerSigner(Signer):
+    def __init__(self, path: str, app: CosmosApp):
+        self.cosmos_app = app if app is not None else CosmosApp(bip44_to_list(path))
+        super().__init__()
+
+    def get_public_key(self) -> PublicKey:
+        return PublicKey.from_hex(self.cosmos_app.ins_get_addr_secp256k1("band", False).public_key)
```

Comment:
I think receive hrp when init and has default value to be `BECH32_ADDR_ACC_PREFIX`